### PR TITLE
build: libevent 2.1.12-stable

### PR DESCRIPTION
**Pull request by fanquake**
Start using [libevent 2.1.12-stable](https://github.com/libevent/libevent/releases/tag/release-2.1.12-stable) in depends. Changelog is available here: https://raw.githubusercontent.com/libevent/libevent/release-2.1.12-stable/ChangeLog. This release basically only contains bug fixes.

Upgrading will also prevent some oss-fuzz reports which are coming from libevent code, such as https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=34355 & https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=34412.

Guix builds:
```bash
aa365feab851d2055c4c3cd787fc1db972a8f2efe1e7c75bc62b34594846a021  guix-build-f152c1ac66bf/output/aarch64-linux-gnu/bitcoin-f152c1ac66bf-aarch64-linux-gnu-debug.tar.gz
80cf96c4618279f4420b3f15ce3d3331937506e7996d8c325ce741f4da431fad  guix-build-f152c1ac66bf/output/aarch64-linux-gnu/bitcoin-f152c1ac66bf-aarch64-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/aarch64-linux-gnu/inputs.SHA256SUMS
b6261b19ed8f55e655122d515a4f76a984e1f74f932630ce40ca34bb76fadfc6  guix-build-f152c1ac66bf/output/arm-linux-gnueabihf/bitcoin-f152c1ac66bf-arm-linux-gnueabihf-debug.tar.gz
be1018237e5e739a742f777efdbdd28ff794cfc2faa39930d5da6dfcfa620da3  guix-build-f152c1ac66bf/output/arm-linux-gnueabihf/bitcoin-f152c1ac66bf-arm-linux-gnueabihf.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/arm-linux-gnueabihf/inputs.SHA256SUMS
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855  guix-build-f152c1ac66bf/output/dist-archive/SKIPATTEST.TAG
52fa25961965b53e73319f5f8ba5a62f789b56ff3a07286ed5fe88771c2674bf  guix-build-f152c1ac66bf/output/dist-archive/bitcoin-f152c1ac66bf.tar.gz
069dfc72b83527722552d12db86f3f7046be9a85d0ccc5168c71af0d38dbc9a9  guix-build-f152c1ac66bf/output/powerpc64-linux-gnu/bitcoin-f152c1ac66bf-powerpc64-linux-gnu-debug.tar.gz
dba47d56dccdd4183762f888ab2e64bf8c074a6873dcf6dac663f88633932926  guix-build-f152c1ac66bf/output/powerpc64-linux-gnu/bitcoin-f152c1ac66bf-powerpc64-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/powerpc64-linux-gnu/inputs.SHA256SUMS
e43e2c631ed0dd587a52705b36e5e69e9e622d493eb6741d438976197a1416f2  guix-build-f152c1ac66bf/output/powerpc64le-linux-gnu/bitcoin-f152c1ac66bf-powerpc64le-linux-gnu-debug.tar.gz
f47f006c68262e3cd30acc355478d4ea056bb1c055bc51c5adab56305c260725  guix-build-f152c1ac66bf/output/powerpc64le-linux-gnu/bitcoin-f152c1ac66bf-powerpc64le-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/powerpc64le-linux-gnu/inputs.SHA256SUMS
f62805fd8652e28d935da14b0328f62a583be24b5517a5c7f189970bdd3c6eb0  guix-build-f152c1ac66bf/output/riscv64-linux-gnu/bitcoin-f152c1ac66bf-riscv64-linux-gnu-debug.tar.gz
db16cbddfc790798af8617213cc726002545f05cfb90ec00e462ba5d3aca10b1  guix-build-f152c1ac66bf/output/riscv64-linux-gnu/bitcoin-f152c1ac66bf-riscv64-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/riscv64-linux-gnu/inputs.SHA256SUMS
2c1afbc6ea44e5947ffcbdf031233f03f7a63b93ca08ddfec2c45c661c1666bc  guix-build-f152c1ac66bf/output/x86_64-apple-darwin18/bitcoin-f152c1ac66bf-osx-unsigned.dmg
65cff5a1bc7eb411b1bb26eb407c2a944e5bf24b08cfcd7a9eeea47924106ac8  guix-build-f152c1ac66bf/output/x86_64-apple-darwin18/bitcoin-f152c1ac66bf-osx-unsigned.tar.gz
98a5c002aed8a35912853cca5e5ffd7afc91eb839b81ca8bc73ddf010839e3ac  guix-build-f152c1ac66bf/output/x86_64-apple-darwin18/bitcoin-f152c1ac66bf-osx64.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/x86_64-apple-darwin18/inputs.SHA256SUMS
84e3188d369aae01a459b477e1935a9466338bb9ea6d4beb93675275a5ac62fd  guix-build-f152c1ac66bf/output/x86_64-linux-gnu/bitcoin-f152c1ac66bf-x86_64-linux-gnu-debug.tar.gz
ea3f4d928f82ea3d1079be3d4e020349e9d412c621149656ea409a61de906c4c  guix-build-f152c1ac66bf/output/x86_64-linux-gnu/bitcoin-f152c1ac66bf-x86_64-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/x86_64-linux-gnu/inputs.SHA256SUMS
6dc14c4e1d0b353b865a9f807dafe7ec56e829a8bc104a5ac6ef331bdc9bc726  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/bitcoin-f152c1ac66bf-win-unsigned.tar.gz
bd382152587c2dc5816d615d71e36df80f54eac5b4913569148e44117b80517c  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/bitcoin-f152c1ac66bf-win64-debug.zip
d2f38b52ea54e913ccadbda6c264a258ca9a55b40a8af96ab8123e8311170aef  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/bitcoin-f152c1ac66bf-win64-setup-unsigned.exe
8846b6edcdb0b3c29bdbab08b3582ee13b4877724bf3811246360b14b73ae420  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/bitcoin-f152c1ac66bf-win64.zip
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/inputs.SHA256SUMS
```

Gitian builds:
```bash
# macOS:
19401d11d0eda5cdcbab8c1353f7110963288e5223fd95d8cc3380107b66a7e4  bitcoin-f152c1ac66bf-osx-unsigned.dmg
3eb2383c5af734a86eaf1895f261c1ba5fddc130d9682857eff08fa7d0599e93  bitcoin-f152c1ac66bf-osx-unsigned.tar.gz
c2572b6f17613a380cf95417ed6a533d8f75fb1418428e234cb90cc0d3d5a655  bitcoin-f152c1ac66bf-osx64.tar.gz
52fa25961965b53e73319f5f8ba5a62f789b56ff3a07286ed5fe88771c2674bf  src/bitcoin-f152c1ac66bf.tar.gz
4bcf84076345c61736c63f33e6f9db57b626030c417c2a1b668752d5f87dab22  bitcoin-core-osx-22-res.yml

# Windows:
5857fda233fc9fdb04391bd1718522b774978d093912650d4dc6e49b70dde40b  bitcoin-f152c1ac66bf-win-unsigned.tar.gz
beff745cde9f19d3d619c3c360a9eb37dbe94169cbc968ac7958e80521584787  bitcoin-f152c1ac66bf-win64-debug.zip
e92010f76d8b286e291c4420349018029e72013ae8bf5b98eccfbf9f856a1490  bitcoin-f152c1ac66bf-win64-setup-unsigned.exe
46c38498fb16e5353de24caffd11d00b23162ec92f126de57888b4f92f375d79  bitcoin-f152c1ac66bf-win64.zip
52fa25961965b53e73319f5f8ba5a62f789b56ff3a07286ed5fe88771c2674bf  src/bitcoin-f152c1ac66bf.tar.gz
851ebc1f3028ec421d733a15acf7514474a44837b75e2f94c1f80ab383b1a364  bitcoin-core-win-22-res.yml

# Linux:
4270014b016a309b6ccff39cf87b90f3b5a36a10eab5d2c41697b12b9eb5e63e  bitcoin-f152c1ac66bf-aarch64-linux-gnu-debug.tar.gz
f226aaeb9cf7786fa5f4dc379611c8b1b3b580df503f79905158d0e2271c3862  bitcoin-f152c1ac66bf-aarch64-linux-gnu.tar.gz
4cd6208dc30c15b71d80d9cd59d367aba676934609d59581c8f418f9e864994d  bitcoin-f152c1ac66bf-arm-linux-gnueabihf-debug.tar.gz
21746b2bbf4fd6778ea0357271f569975029667dea2ad0000abb6751eeba8071  bitcoin-f152c1ac66bf-arm-linux-gnueabihf.tar.gz
d73e6b2a779061037eaa3174615a7e5bb6082056ee4dc13de4026cc3ad21f3c5  bitcoin-f152c1ac66bf-powerpc64-linux-gnu-debug.tar.gz
e6c5876dcdd30b8936b957afc8adb9341c64a1201ba8c4285f5f76bc37689c7c  bitcoin-f152c1ac66bf-powerpc64-linux-gnu.tar.gz
fc0270f7c220519e91ecd27c37df7ec130e16db8508e51b52eb88eae1f0d05d2  bitcoin-f152c1ac66bf-powerpc64le-linux-gnu-debug.tar.gz
3729a36717f704a6e30a0368a68d2356ec9af240b706a7a8c117f7ebcc5ebbca  bitcoin-f152c1ac66bf-powerpc64le-linux-gnu.tar.gz
b661fc47dbf4a56cecfe02abfff78644524cd6d94aac51a38d83a4ba96a21b6e  bitcoin-f152c1ac66bf-riscv64-linux-gnu-debug.tar.gz
8934ae4727e7908f626d2392877cad153696c6d8e832a9dbc0960e090f633ab4  bitcoin-f152c1ac66bf-riscv64-linux-gnu.tar.gz
cc8511b71c89575f15cfbb360c18c0af298869fd4e8c4a27ef2c04b6bc0f0703  bitcoin-f152c1ac66bf-x86_64-linux-gnu-debug.tar.gz
1cfb8cf750ca7af47cbab00d8eaee336fb11c22756a62b26b5c9a1263b5b0cd4  bitcoin-f152c1ac66bf-x86_64-linux-gnu.tar.gz
52fa25961965b53e73319f5f8ba5a62f789b56ff3a07286ed5fe88771c2674bf  src/bitcoin-f152c1ac66bf.tar.gz
3b5053c7392a1a6264f3d192136e50fc9db969834d9d3643d860a3f0e2f79535  bitcoin-core-linux-22-res.yml
```

### CODE REVIEW

**Comment by hebasto** [comment]
> Changelog is available here: https://raw.githubusercontent.com/libevent/libevent/release-2.1.12-stable/ChangeLog.

A cannot find the reason why the patch could be dropped?
And the related https://github.com/libevent/libevent/issues/1041 is still open, btw.

**Comment by fanquake** [comment]
> Cannot find the reason why the patch could be dropped?

This is the relevant changelog entry:
>  o autoconf: fix getaddrinfo checking errors on mingw (b9bf7fa7 yuangongji)

which is [this commit](https://github.com/libevent/libevent/commit/6d54be2cc078351b4ce1e469bedc4a976f4e3636). The patch is small, so it's easy to verify that the changes are part of the newer libevent source.

**Comment by practicalswift** [comment]
cr ACK f152c1ac66bfe5c77d947d5c99ce5da3a88797fa: patch looks correct and this should resolve the OSS-Fuzz issues

File: depends/packages/libevent.mk [hunk 1]
```diff
@@ -1,14 +1,8 @@
 package=libevent
-$(package)_version=2.1.11-stable
-$(package)_download_path=https://github.com/libevent/libevent/archive/
-$(package)_file_name=release-$($(package)_version).tar.gz
-$(package)_sha256_hash=229393ab2bf0dc94694f21836846b424f3532585bac3468738b7bf752c03901e
-$(package)_patches=0001-fix-windows-getaddrinfo.patch
-
-define $(package)_preprocess_cmds
-   patch -p1 < $($(package)_patch_dir)/0001-fix-windows-getaddrinfo.patch && \
-  ./autogen.sh
```
[the remark this answers is not in the document]
Comment (hebasto) [reply depends/packages/libevent.mk:10]: Oh, I see the `configure` has already been generated in the release archive.

**Review by hebasto — APPROVED** [review]
~ACK f152c1ac66bfe5c77d947d5c99ce5da3a88797fa~ [retracted due to non-reproducible builds](https://github.com/bitcoin/bitcoin/pull/21991#issuecomment-844005749)

On master (6d1d33d33491a98bb0dbf64ea7e4743200e71474):
```
$ make -C depends libevent_configured HOST=x86_64-w64-mingw32 2>&1 | grep getaddrinfo
checking for getaddrinfo... yes
```

With this PR (f152c1ac66bfe5c77d947d5c99ce5da3a88797fa):
```
$ make -C depends libevent_configured HOST=x86_64-w64-mingw32 2>&1 | grep getaddrinfo
checking whether getaddrinfo is declared... (cached) yes
```

**Comment by hebasto** [comment]
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
cfecdf7929c733d811078b887e3c1305d0ec1bf8c9fbf017d4df5aff8d835e96  guix-build-f152c1ac66bf/output/aarch64-linux-gnu/bitcoin-f152c1ac66bf-aarch64-linux-gnu-debug.tar.gz
04b72765deb872600be2143a27f1f3d8f40972621bc64fa3658394db9d1a1164  guix-build-f152c1ac66bf/output/aarch64-linux-gnu/bitcoin-f152c1ac66bf-aarch64-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/aarch64-linux-gnu/inputs.SHA256SUMS
e46b551c256058b6e0af3bd4fc0f30058ee6c44a1ef9dcc847aebe59019d7a25  guix-build-f152c1ac66bf/output/arm-linux-gnueabihf/bitcoin-f152c1ac66bf-arm-linux-gnueabihf-debug.tar.gz
0bc4c49eb2f3faba82aaffe6a61e06632a5c25c800aad1bdc3dfcfe5f1dc4769  guix-build-f152c1ac66bf/output/arm-linux-gnueabihf/bitcoin-f152c1ac66bf-arm-linux-gnueabihf.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/arm-linux-gnueabihf/inputs.SHA256SUMS
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855  guix-build-f152c1ac66bf/output/dist-archive/SKIPATTEST.TAG
52fa25961965b53e73319f5f8ba5a62f789b56ff3a07286ed5fe88771c2674bf  guix-build-f152c1ac66bf/output/dist-archive/bitcoin-f152c1ac66bf.tar.gz
1f28e6a03908f557ee39b3a7f3319b954846ad9ff167c734a36bfb214cc31603  guix-build-f152c1ac66bf/output/powerpc64-linux-gnu/bitcoin-f152c1ac66bf-powerpc64-linux-gnu-debug.tar.gz
e18613841275d4ec54e810a9dbed4ab8cab02d1d9906d52dc09a9f6fc1492436  guix-build-f152c1ac66bf/output/powerpc64-linux-gnu/bitcoin-f152c1ac66bf-powerpc64-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/powerpc64-linux-gnu/inputs.SHA256SUMS
5834c44b2e8f8e2592828b6e13b6e6a423cc874d3f16f2534bff582d45437f4e  guix-build-f152c1ac66bf/output/powerpc64le-linux-gnu/bitcoin-f152c1ac66bf-powerpc64le-linux-gnu-debug.tar.gz
a8b3fd77a6f43e3cbbed0e0100e35e946d5439ca662a2a74e6c99e03086ac2ca  guix-build-f152c1ac66bf/output/powerpc64le-linux-gnu/bitcoin-f152c1ac66bf-powerpc64le-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/powerpc64le-linux-gnu/inputs.SHA256SUMS
f3248901ef94bf024aa9489266a13f00e2cba717c6ba8f8be50dfcdbc0babd38  guix-build-f152c1ac66bf/output/riscv64-linux-gnu/bitcoin-f152c1ac66bf-riscv64-linux-gnu-debug.tar.gz
e74bc114c6ef5304105b54296b9308d1f5b840ff006c715264183a5db5a44d95  guix-build-f152c1ac66bf/output/riscv64-linux-gnu/bitcoin-f152c1ac66bf-riscv64-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/riscv64-linux-gnu/inputs.SHA256SUMS
2c1afbc6ea44e5947ffcbdf031233f03f7a63b93ca08ddfec2c45c661c1666bc  guix-build-f152c1ac66bf/output/x86_64-apple-darwin18/bitcoin-f152c1ac66bf-osx-unsigned.dmg
65cff5a1bc7eb411b1bb26eb407c2a944e5bf24b08cfcd7a9eeea47924106ac8  guix-build-f152c1ac66bf/output/x86_64-apple-darwin18/bitcoin-f152c1ac66bf-osx-unsigned.tar.gz
98a5c002aed8a35912853cca5e5ffd7afc91eb839b81ca8bc73ddf010839e3ac  guix-build-f152c1ac66bf/output/x86_64-apple-darwin18/bitcoin-f152c1ac66bf-osx64.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/x86_64-apple-darwin18/inputs.SHA256SUMS
864087cf6e2c6bf1d140d25fe50cc2c510526a698bfc89d3b01ae787dbd8e90e  guix-build-f152c1ac66bf/output/x86_64-linux-gnu/bitcoin-f152c1ac66bf-x86_64-linux-gnu-debug.tar.gz
1013bc16b69fadaba6ee52a75ad5db8fe15d7448b67a6f11a81dd62694d35e3a  guix-build-f152c1ac66bf/output/x86_64-linux-gnu/bitcoin-f152c1ac66bf-x86_64-linux-gnu.tar.gz
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/x86_64-linux-gnu/inputs.SHA256SUMS
9530fab347f3ac40b3dcbfab8702af7b81d0cea6188ff44a01169ca97a7250e5  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/bitcoin-f152c1ac66bf-win-unsigned.tar.gz
4cba6759439d76708390e44e89b8fb6df275a502e331780e3a056d1035635dfb  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/bitcoin-f152c1ac66bf-win64-debug.zip
db43d7263f8bd54690b76c20268684de506124b49a7cb61753a4ecf524e7ea2e  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/bitcoin-f152c1ac66bf-win64-setup-unsigned.exe
5ceffabc4e8c2cd1258155d04e74ca24632a21e29c3d1d85e3c52d3d9e58502d  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/bitcoin-f152c1ac66bf-win64.zip
53db9e6b78fa13e0753cab4a35da5e66156c54485020c6c49b1cd1ab406e978b  guix-build-f152c1ac66bf/output/x86_64-w64-mingw32/inputs.SHA256SUMS
```

Hashes differ from ones in the OP.

(retracting my ACK for now)

**Comment by laanwj** [comment]
Concept ACK, libevent changes in the changelog look fine to me. Lots of small fixes of which I'm not sure if they affect our use or not, but can't hurt either.

**Comment by hebasto** [comment]
Probably, d2ccb92f8bfaa785bab094a02e2d858d55279930 could fix guix build determinism.

**Comment by MarcoFalke** [comment]
Concept ACK f152c1ac66bfe5c77d947d5c99ce5da3a88797fa

**Comment by fanquake** [comment]
We've received another oss-fuzz report for libevent: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=34509. Note that all 3 linked here are the same underlying issue.

Given the determinism issue raised is not related to this PR, I'm still happy for this to merged as is.

**Pull request merged**